universal_transfer_operator/src/universal_transfer_operator/data_providers/database/snowflake.py

```diff
@@ -0,0 +1,264 @@
+        return self.load_file_to_table(
+            input_file=source_ref.actual_file,
+            output_table=self.dataset,
+        )
+
+    # ---------------------------------------------------------
+    # Table metadata
+    # ---------------------------------------------------------
+    @staticmethod
+    def get_table_qualified_name(table: Table) -> str:  # skipcq: PYL-R0201
+        """
+        Return table qualified name. In Snowflake, it is the database, schema and table
+
+        :param table: The table we want to retrieve the qualified name for.
+        """
+        qualified_name_lists = [
+            table.metadata.database,
+            table.metadata.schema,
+            table.name,
+        ]
+        qualified_name = ".".join(name for name in qualified_name_lists if name)
+        return qualified_name
+
+    def schema_exists(self, schema: str) -> bool:
+        """
+        Checks if a schema exists in the database
+
+        :param schema: DB Schema - a namespace that contains named objects like (tables, functions, etc)
+        """
+
+        # Below code is added due to breaking change in apache-airflow-providers-snowflake==3.2.0,
+        # we need to pass handler param to get the rows. But in version apache-airflow-providers-snowflake==3.1.0
+        # if we pass the handler provider raises an exception AttributeError 'sfid'.
+        try:
+            schemas = self.hook.run(
+                "SELECT SCHEMA_NAME from information_schema.schemata WHERE LOWER(SCHEMA_NAME) = %(schema_name)s;",
+                parameters={"schema_name": schema.lower()},
+                handler=lambda cur: cur.fetchall(),
+            )
+        except AttributeError:
+            schemas = self.hook.run(
+                "SELECT SCHEMA_NAME from information_schema.schemata WHERE LOWER(SCHEMA_NAME) = %(schema_name)s;",
+                parameters={"schema_name": schema.lower()},
+            )
+        try:
+            # Handle case for apache-airflow-providers-snowflake<4.0.1
+            created_schemas = [x["SCHEMA_NAME"] for x in schemas]
+        except TypeError:
+            # Handle case for apache-airflow-providers-snowflake>=4.0.1
+            created_schemas = [x[0] for x in schemas]
+        return len(created_schemas) == 1
+
+    def create_table_using_schema_autodetection(
+        self,
+        table: Table,
+        file: File | None = None,
+        dataframe: pd.DataFrame | None = None,
+        columns_names_capitalization: ColumnCapitalization = "original",
+    ) -> None:  # skipcq PYL-W0613
+        """
+        Create a SQL table, automatically inferring the schema using the given file.
+        Overriding default behaviour and not using the `prep_table` since it doesn't allow the adding quotes.
+
+        :param table: The table to be created.
+        :param file: File used to infer the new table columns.
+        :param dataframe: Dataframe used to infer the new table columns if there is no file
```

Comment:
please add :param for columns_names_capitalization

Reply:
added param